### PR TITLE
[Bug 1169416] Show Translation page now shows full locale name along …

### PR DESCRIPTION
**Pull request by skiee-mahir**

### CODE REVIEW

**Comment by safwanrahman** [comment]
hmm. there is a bug in your code!

**Comment by mythmon** [comment]
@skiee-mahir: This change has broken the tests. You can see the failure [in our CI logs](https://travis-ci.org/mozilla/kitsune/jobs/78089754#L1921). I encourage you to [set up the tests locally](http://kitsune.readthedocs.org/en/latest/tests.html#tests-chapter) so you can run the tests on your own system while debugging this. It makes working on the problem a lot easier.

File: kitsune/wiki/views.py [hunk 1]
```diff
@@ -1327,14 +1327,18 @@ def show_translations(request, document_slug):
...
-    translated_locales.extend(document.translations.all().values_list(
+    translated_locales_code.append(document.locale)
+    translated_locales_code.extend(document.translations.all().values_list(
         'locale', flat=True))
-
+    # Seperating Translated and Untranslated locales
     for locale in settings.LANGUAGE_CHOICES:
-        if not locale[0] in translated_locales:
-            untranslated_locales.append(locale[0])
+        if not locale[0] in translated_locales_code:
```
Comment (safwanrahman) [inline kitsune/wiki/views.py]: Here seems to be the bug! :+1:

**Comment by safwanrahman** [comment]
I have tested your patch, and it looks like this.
As you can see, that the ``en-US`` is gone down to bottom, where it should be at top always. Make your code works like that it be at top always.
firefox_screenshot_2015-08-31t23-16-40 855z

**Comment by safwanrahman** [comment]
Then, you need to change here to apply the new changes. as In past the locale code would only show, as your patch makes the locale name also be showed, you need to change in the test to test that.

here it is:
https://github.com/mozilla/kitsune/blob/master/kitsune/wiki/tests/test_templates.py#L2106-L2107

**Comment by skiee-mahir** [comment]
0 static files copied to '/home/xren/Desktop/kitsune/static', 2349 unmodified, 267 post-processed.
Creating test database for alias 'default'...
Got an error creating the test database: (1044, "Access denied for user 'kitsune'@'localhost' to database 'test_kitsune'")
Type 'yes' if you would like to try deleting the test database 'test_kitsune', or 'no' to cancel: 


Any clues how to fix this?

**Comment by safwanrahman** [comment]
Did you read the documentation? check the doocumentation to get how to set up test! 
or you can ping me at IRC

**Comment by mythmon** [comment]
Sounds like you didn't correctly set up permissions for the testing database. The tests use a separate database, since they want to start with a fresh version of the database. You'll have to add a separate grant for that. Sorry that wasn't in the docs I linked you before. [Instructions for that are here](http://kitsune.readthedocs.org/en/latest/hacking_howto.html?highlight=test_kitsune#setting-it-up).

**Comment by safwanrahman** [comment]
@skiee-mahir I have found a way!
if you import ``LOCALES from kitsune.lib.sumo_locales`` and then get ``LOCALES[locale_code].native``, you can get the full locale name of that locale code. I think you can now fix this bug more flexibly. Let me know If I can do any!

**Comment by safwanrahman** [comment]
@skiee-mahir And at the end, you must need to change some in the test. Fix the above one and then we will together make the test fixed! :+1: :smile:

**Comment by skiee-mahir** [comment]
yeah, I noticed that test looks for "en-US" as the text but we have "English (en-US)" :wink:

**Comment by mythmon** [comment]
This PR isn't passing the tests, and hasn't been updated in over 3 weeks. I'm going to close it to reduce the PR queue. If you work more on this, please feel free to re-open this PR or file a new one.

**Comment by skiee-mahir** [comment]
@safwanrahman I don't actually understand how you plan to implement that piece of code? In template? If not in Template, then with LOCALES["locale"].native() we'll be sending out full locale name to the template which will mess the links really bad :(

**Comment by safwanrahman** [comment]
@skiee-mahir, I have write up some code in the Django shell to check how to get the work done. Please check the code below and see how I arranged to get the "bn-BD" locale at the first of the list. you should work like this in order to get "en-US" at first of the list


```python
>>>from kitsune.lib.sumo_locales import LOCALES
>>>bn_bd_full = LOCALES["bn-BD"].native
>>>bn_bd = ("bn-BD", bn_bd_full)
>>>en_us = ("en-US", "English")
>>>all_locales = []
>>>all_locales.append(bn_bd)
>>>all_locales.append(en_us)
>>>print all_locales
>>> [('bn-BD', u'\u09ac\u09be\u0982\u09b2\u09be (\u09ac\u09be\u0982\u09b2\u09be\u09a6\u09c7\u09b6)'), ('en-US', 'English')]
```

**Comment by skiee-mahir** [comment]
@safwanrahman check [this commit](https://github.com/skiee-mahir/kitsune/commit/bb75c22e788e9f50b85bab3f7e946a5f2ef9e32b ). I think this is what you wanted. 

@mythmon we need to fix the tests right, [here](https://github.com/mozilla/kitsune/blob/master/kitsune/wiki/tests/test_templates.py#L2106-L2107). This seeks for "en-US" at the top, but to fix this bug, we need to place "English (en-US)".

**Comment by safwanrahman** [comment]
@skiee-mahir, Can you please make the commit in the same branch of this PR? so it will be easier to review.

**Comment by skiee-mahir** [comment]
@safwanrahman it's in the same branch! But the PR is closed and I don't see any __reopen button__, it's *wired* .

**Comment by mythmon** [comment]
The re-open button should be at the bottom, next to the comment button. Maybe there is something about GitHub permissions preventing you from doing that. I'll re-open the PR.

**Comment by skiee-mahir** [comment]
@mythmon @safwanrahman  can you help me understand what is this [line](https://github.com/mozilla/kitsune/blob/master/kitsune/wiki/tests/test_templates.py#L2107) for?

**Comment by skiee-mahir** [comment]
Due to some some serious conflicts, which I couldn't handle, I'm closing this! The new pull is, [here](https://github.com/mozilla/kitsune/pull/2673).

**Pull request closed without merging**